ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -0,0 +1,344 @@
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests for {@link PersistentIndex}. Tests both segmented and non segmented log use cases.
+ */
+@RunWith(Parameterized.class)
+public class BlobStoreStatsTest {
+  private static final StoreKeyFactory STORE_KEY_FACTORY;
+  private static final byte[] RECOVERY_INFO = new byte[100];
+
+  static {
+    try {
+      STORE_KEY_FACTORY = Utils.getObj("com.github.ambry.store.MockIdFactory");
+      Arrays.fill(RECOVERY_INFO, (byte) 0);
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  private static final int BUCKET_COUNT = 200;
+  private static final long BUCKET_TIME_SPAN = 1000;
+  private static final long SEGMENT_SCAN_OFFSET = 100000;
+
+  private final CuratedLogIndexState state;
+  // Variables that represent the folder where the data resides
+  private final File tempDir;
+
+  // the BlobStoreStats that exposes statistics about the blobStore
+  private final BlobStoreStats blobStoreStats;
+
+  /**
+   * Running for both segmented and non-segmented log.
+   * @return an array with both {@code false} and {@code true}.
+   */
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    return Arrays.asList(new Object[][]{{false}, {true}});
+  }
+
+  /**
+   * Creates a temporary directory and sets up some test state.
+   * @throws IOException
+   */
+  public BlobStoreStatsTest(boolean isLogSegmented) throws InterruptedException, IOException, StoreException {
+    tempDir = StoreTestUtils.createTempDirectory("indexDir-" + UtilsTest.getRandomString(10));
+    state = new CuratedLogIndexState(isLogSegmented, tempDir);
+    blobStoreStats = new BlobStoreStats(state.log, state.index, BUCKET_COUNT, BUCKET_TIME_SPAN, SEGMENT_SCAN_OFFSET,
+        false, state.time);
+  }
+
+  /**
+   * Releases all resources and deletes the temporary directory.
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @After
+  public void cleanup() throws InterruptedException, IOException, StoreException {
+    state.destroy();
+    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+  }
+
+  @Test
+  public void testTotalCapacity() throws IOException, StoreException, InterruptedException {
+    // Verify total capacity
+    assertEquals("Total capacity mismatch ", state.getTotalCapcity(), blobStoreStats.getTotalCapacity());
+    // Total capacity should be unaffected by any new writes to the store
+    state.addPutEntries(10, state.PUT_RECORD_SIZE, Utils.Infinite_Time);
+    assertEquals("Total capacity mismatch ", state.getTotalCapcity(), blobStoreStats.getTotalCapacity());
+  }
+
+  @Test
+  public void testUsedCapacity() throws IOException, StoreException, InterruptedException {
```

Comment:
docs for all tests - describing what it is testing and the test methodology if it is not simple.